### PR TITLE
If DEFAULT_USE_PROMETHEUS_UWSGI is true, run uwsgi_exporter when use_…

### PR DESCRIPTION
**Pull request by EvanKrall**
…prometheus is unspecified.

### CODE REVIEW

**Review by jfongatyelp — APPROVED** [review]

File: tests/test_kubernetes_tools.py [hunk 1]
```diff
@@ -767,18 +767,43 @@ def test_should_run_uwsgi_exporter_sidecar_defaults(self):
...
+                )
+                is False
+            )
+
+        # If the default for use_prometheus is True and config_dict doesn't specify use_prometheus, we should run
+        # uwsgi_exporter regardless of default_should_run_uwsgi_exporter_sidecar.
+        with mock.patch(
+            "paasta_tools.kubernetes_tools.DEFAULT_USE_PROMETHEUS_UWSGI",
+            autospec=True,
+            new=True,
```
Comment (nemacysts) [inline tests/test_kubernetes_tools.py:793]:
```suggestion
```

File: tests/test_kubernetes_tools.py [hunk 2]
```diff
@@ -767,18 +767,43 @@ def test_should_run_uwsgi_exporter_sidecar_defaults(self):
             default_should_run_uwsgi_exporter_sidecar=mock.Mock(return_value=False)
         )
 
-        assert (
-            self.deployment.should_run_uwsgi_exporter_sidecar(
-                system_paasta_config_enabled
+        with mock.patch(
+            "paasta_tools.kubernetes_tools.DEFAULT_USE_PROMETHEUS_UWSGI",
+            autospec=True,
+            new=False,
```
Comment (nemacysts) [inline tests/test_kubernetes_tools.py:773]:
```suggestion
```

**Review by nemacysts — APPROVED** [review]

File: tests/test_kubernetes_tools.py (hunk 1 above)
Comment (EvanKrall) [reply tests/test_kubernetes_tools.py:793]: ? These lines are important to the test. It's what's setting DEFAULT_USE_PROMETHEUS_UWSGI to True and False.

Comment (EvanKrall) [reply tests/test_kubernetes_tools.py:793]: (oh but I did forget to set autospec=False after the pre-commit complained about me not having an autospec line)

Comment (nemacysts) [reply tests/test_kubernetes_tools.py:793]: ah, I didn't read the error message better than the tests and suggested removing the wrong line :p

**Review by nemacysts — APPROVED** [review]

**Pull request merged**